File: 0048-rotate-image/0048-rotate-image.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        for i in range(len(matrix)):
            for j in range(len(matrix)):
                if j - i > 0:
                    matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]

        for row in matrix:
            row.reverse()

File: 0048-rotate-image/0048-rotate-image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int n = matrix.size();
        
        // Step 1: Swap rows vertically (reverse the rows)
        int l = 0, r = n - 1;
        while (l < r) {
            swap(matrix[l], matrix[r]);
            l++;
            r--;
        }

        // Step 2: Transpose the matrix by swapping across the diagonal
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {  // Start j from i+1 to avoid redundant swaps
                swap(matrix[i][j], matrix[j][i]);
            }
        }
    }
};
